packages/angular/src/generators/ng-add/utilities/workspace.ts

```diff
@@ -0,0 +1,350 @@
+      json.scripts.postinstall &&
+      !json.scripts.postinstall.includes('decorate-angular-cli.js')
+    ) {
+      // if exists, add execution of this script
+      json.scripts.postinstall += ' && node ./decorate-angular-cli.js';
+    } else {
+      if (!json.scripts) json.scripts = {};
+      // if doesn't exist, set to execute this script
+      json.scripts.postinstall = 'node ./decorate-angular-cli.js';
+    }
+    if (json.scripts.ng) {
+      json.scripts.ng = 'nx';
+    }
+    return json;
+  });
+}
+
+export function updateWorkspaceConfigDefaults(tree: Tree): void {
+  const workspaceConfig = readWorkspaceConfiguration(tree);
+  delete (workspaceConfig as any).newProjectRoot;
+  workspaceConfig.cli = workspaceConfig.cli ?? {};
+  if (!workspaceConfig.cli.defaultCollection) {
+    workspaceConfig.cli.defaultCollection = '@nrwl/angular';
+  }
+  updateWorkspaceConfiguration(tree, workspaceConfig);
+}
+
+export function updateRootTsConfig(tree: Tree): void {
+  writeJson(
+    tree,
+    'tsconfig.base.json',
+    setUpCompilerOptions(readJson(tree, getRootTsConfigPathInTree(tree)))
+  );
+  if (tree.exists('tsconfig.json')) {
+    tree.delete('tsconfig.json');
+  }
+}
+
+export function updatePackageJson(tree: Tree): void {
+  updateJson(tree, 'package.json', (packageJson) => {
+    packageJson.scripts = packageJson.scripts || {};
+    packageJson.scripts = {
+      ...packageJson.scripts,
+      nx: 'nx',
+      'affected:apps': 'nx affected:apps',
+      'affected:libs': 'nx affected:libs',
+      'affected:build': 'nx affected:build',
+      'affected:e2e': 'nx affected:e2e',
+      'affected:test': 'nx affected:test',
+      'affected:lint': 'nx affected:lint',
+      'affected:graph': 'nx affected:graph',
+      affected: 'nx affected',
+      format: 'nx format:write',
+      'format:write': 'nx format:write',
+      'format:check': 'nx format:check',
+      update: 'ng update @nrwl/workspace',
+      'update:check': 'ng update',
+      lint: 'nx workspace-lint && ng lint',
+      graph: 'nx graph',
+      'workspace-generator': 'nx workspace-generator',
+      help: 'nx help',
+    };
+    packageJson.devDependencies = packageJson.devDependencies ?? {};
+    packageJson.dependencies = packageJson.dependencies ?? {};
+    if (!packageJson.dependencies['@nrwl/angular']) {
+      packageJson.dependencies['@nrwl/angular'] = nxVersion;
+    }
+    if (!packageJson.devDependencies['@angular/cli']) {
+      packageJson.devDependencies['@angular/cli'] = angularDevkitVersion;
+    }
+    if (!packageJson.devDependencies['@nrwl/workspace']) {
+      packageJson.devDependencies['@nrwl/workspace'] = nxVersion;
+    }
+    if (!packageJson.devDependencies['nx']) {
+      packageJson.devDependencies['nx'] = nxVersion;
+    }
+    if (!packageJson.devDependencies['prettier']) {
+      packageJson.devDependencies['prettier'] = prettierVersion;
+    }
+
+    return packageJson;
+  });
+}
+
+export function updateTsLint(tree: Tree): void {
```

Comment:
If they have TSLint, do we want to migrate them to ESLint, or just keep them supported?

Reply:
Yeah, it's something I've considered doing, offering the dev to migrate, wouldn't do it automatically. As with other changes, it's something that I'll be looking into follow-up PRs, this one is only a refactor, no functionality was changed.